uplink/builder.py

```diff
@@ -85,9 +85,16 @@ def execute(self):
 
 class Builder(interfaces.AbstractUplinkBuilder):
 
-    def __init__(self, service_cls):
+    def __init__(self, service_stub):
+        """
+        Creates a Builder.
+
+        :param object service_stub: An initiated instance of the class being built, with
+                                    no functions or fields
+        """
+
         self._base_url = ""
-        self._service_cls = service_cls
+        self._service_stub = service_stub
```

Comment:
I guess since now that we have the `Consumer` interface, we might want to consider keeping the `Builder` as an implementation detail once we move it out of the public API.  

For example, the `Builder.build` method could accept a`uplink.interfaces.AbstractRequestDefinition` instance and return a `uplink.builder.CallFactory`. Sounds intriguing, but I'm getting ahead of myself and haven't really thought out the refactoring effort this would require, so we can discuss this in a separate issue and pull request!